needle-notion-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,154 @@
+import type {
+  ConnectorRequest,
+  CreateConnectorRequest,
+} from "~/models/connectors-models";
+import {
+  createConnector,
+  getConnector,
+  listConnectors,
+  deleteConnector,
+  publishConnectorRun,
+  createNeedleFileId,
+  type Session,
+  type ConnectorRunDescriptor,
+} from "@needle-ai/needle-sdk";
+import { db } from "../db";
+import { eq } from "drizzle-orm";
+import { notionConnectorsTable, notionPagesTable } from "../db/schema";
+import { Client as NotionClient } from "@notionhq/client";
+import type {
+  DatabaseObjectResponse,
+  PageObjectResponse,
+} from "@notionhq/client/build/src/api-endpoints";
+
+export async function createNotionConnector(
+  request: CreateConnectorRequest,
+  session: Session,
+) {
+  const connector = await createConnector(
+    {
+      name: request.name,
+      cronJob: request.cronJob,
+      cronJobTimezone: request.cronJobTimezone,
+      collectionIds: [request.collectionId],
+      credentials: request.notionToken.access_token,
+    },
+    session.id,
+  );
+
+  const pagesToInsert = request.notionPages.map((p) => ({
+    ndlConnectorId: connector.id,
+    ndlFileId: createNeedleFileId(),
+    notionUrl: p.url,
+    notionPageId: p.id,
+    notionLastEditedTime: p.last_edited_time,
+  }));
+  await db.insert(notionPagesTable).values(pagesToInsert);
+
+  const connectorsToInsert = request.notionPages.map((p) => ({
```

Comment:
not sure why we duplicate connector rows for each page in this table? We could write only as many rows as workspaces selected. 

`notionConnectors`
connectorId         workspaceId         workspaceName
           1                               X                             Tasks
           1                               Y                          Roadmap
           
The necessary association with `connectorId` for each page comes from the other table `notionPages.connectorId`

Let's simplify this `notionConnectors` table - wdyt?

Reply:
I removed the page references from this table. now there will be only one entry per token.